commit 7fcb91d94e897413c0345bb32ea11293f33efbb1
Author: Nathan Chancellor <nathan@kernel.org>
Date:   Thu Apr 11 11:11:07 2024 -0700

    configs/hardening: Disable CONFIG_UBSAN_SIGNED_WRAP
    
    kernel/configs/hardening.config turns on UBSAN for the bounds sanitizer,
    as that in combination with trapping can stop the exploitation of buffer
    overflows within the kernel. At the same time, hardening.config turns
    off every other UBSAN sanitizer because trapping means all UBSAN reports
    will be fatal and the problems brought up by other sanitizers generally
    do not have security implications.
    
    The signed integer overflow sanitizer was recently added back to the
    kernel and it is default on with just CONFIG_UBSAN=y, meaning that it
    gets enabled when merging hardening.config into another configuration.
    While this sanitizer does have security implications like the array
    bounds sanitizer, work to clean up enough instances to allow this to run
    in production environments is still ramping up, which means regular
    users and testers may be broken by these instances with
    CONFIG_UBSAN_TRAP=y. Disable CONFIG_UBSAN_SIGNED_WRAP in
    hardening.config to avoid this situation.
    
    Fixes: 557f8c582a9b ("ubsan: Reintroduce signed overflow sanitizer")
    Signed-off-by: Nathan Chancellor <nathan@kernel.org>
    Link: https://lore.kernel.org/r/20240411-fix-ubsan-in-hardening-config-v1-2-e0177c80ffaa@kernel.org
    Signed-off-by: Kees Cook <keescook@chromium.org>

diff --git a/kernel/configs/hardening.config b/kernel/configs/hardening.config
index d6f6dc45628a..4b4cfcba3190 100644
--- a/kernel/configs/hardening.config
+++ b/kernel/configs/hardening.config
@@ -41,6 +41,7 @@ CONFIG_UBSAN_BOUNDS=y
 # CONFIG_UBSAN_SHIFT is not set
 # CONFIG_UBSAN_DIV_ZERO is not set
 # CONFIG_UBSAN_UNREACHABLE is not set
+# CONFIG_UBSAN_SIGNED_WRAP is not set
 # CONFIG_UBSAN_BOOL is not set
 # CONFIG_UBSAN_ENUM is not set
 # CONFIG_UBSAN_ALIGNMENT is not set